### PR TITLE
Adding function to display json configuration to readfile. Minor revisions to populateCustomerList().

### PR DESCRIPTION
**Pull request by frostyclawz**
Adding a javascript function to display the configuration as a JSON in the text box. Making minor revisions to populateCustomerList().

### CODE REVIEW

File: src/main/webapp/appscript.js [hunk 1]
```diff
@@ -13,27 +13,49 @@
...
+/**
+ * Fetch a json configuration as a string and formats it to be shown on
+ * the readfile page.
+ */
+async function addToPage() {
+  var vendorID = document.getElementById('customer-ids');
+  var accountID = document.getElementById('account-ids');
+
+  const response = await fetch("/BillingConfig?" + "vendor-id=" 
+    + vendorID + "&account-id=" + accountID);
```
Comment (cadecrandall) [inline src/main/webapp/appscript.js:47]: looks good!

File: src/main/webapp/appscript.js [hunk 2]
```diff
@@ -13,27 +13,49 @@
...
+
+  // Response is a JSON as a string.
+  let config_json = await response.json();
+
+  var html = document.getElementById('json-box');
+
+  // Stringify converts the json into a string.
+  html.innerHTML = JSON.stringify(config_json, undefined, 4);
```
Comment (cadecrandall) [inline src/main/webapp/appscript.js:55]: What do the last two parameters do here?

File: src/main/webapp/readfile.html [hunk 3]
```diff
@@ -15,57 +15,25 @@ <h2>Read Configuration</h2>
       <a href="deletefile.html">Delete Configuration</a>
     </div>
     <div>
-    <form id="config-search" action="/BillingConfig" method="get">
+    <form id="config-search" action="/BillingConfig">
```
Comment (cadecrandall) [inline src/main/webapp/readfile.html:18]: I think this should still use the get method, but I'm not certain

Comment (frostyclawz) [reply src/main/webapp/readfile.html:18]: It should be a get method by default when fetching. But I can change it.

**Comment by frostyclawz** [comment]
Screenshot 2020-07-20 at 2 34 50 PM - Display 2

File: src/main/java/servlets/BillingConfig.java [hunk 4]
```diff
@@ -65,6 +68,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     } else {
       configText = "Error finding " + vendorID + "'s configuration";
     }
+    System.out.println(configText);
```
Comment (cxfloeder) [inline src/main/java/servlets/BillingConfig.java:71]: This can be removed

File: src/main/java/servlets/BillingConfig.java [hunk 5]
```diff
@@ -33,7 +33,8 @@
 
 @WebServlet("/BillingConfig")
 public class BillingConfig extends HttpServlet {
-  private static final String CONTENT_TYPE_TEXT_HTML = "text/html;";
+  // May need to be "application/json;
+  private static final String CONTENT_TYPE_TEXT_HTML = "application/json;";
```
Comment (ianjjohnson) [inline src/main/java/servlets/BillingConfig.java:37]: rm the comment and rename this to CONTENT_TYPE_APPLICATION_JSON

File: src/main/webapp/appscript.js [hunk 6]
```diff
@@ -13,27 +13,73 @@
...
+
+// TODO: Create config and redirect/populate edit form. 
+async function createConfiguration() {
+
+}
 
+// function doGetRequest(var endpointUrl, var params) {
+// if (params != '{}') {
+// var requestString = endpointUrl + "?";
+// for (const [key, value] of Object.entries(params)) {
```
Comment (cxfloeder) [inline src/main/webapp/appscript.js:79]: Why is this all commented out?

File: src/main/java/servlets/BillingConfig.java [hunk 7]
```diff
@@ -56,6 +57,8 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     // GET method --> returns entire billing config as JSON
     // return null if vendor ID doesn't exist
     String vendorID = request.getParameter(VENDOR_ID);
+    vendorID = "sample";
+    // TODO: @cade Add query string functionality to find vendor configs.
```
Comment (ianjjohnson) [inline src/main/java/servlets/BillingConfig.java:61]: nit: put this TODO above the line above it.

File: src/main/webapp/readfile.html [hunk 8]
```diff
@@ -15,57 +15,25 @@ <h2>Read Configuration</h2>
...
       <datalist id="customer-ids">
+        <option value="sample">
       </datalist>
       <p>Account ID:</p>
       <input list="account-ids">
       <datalist id="account-ids">
         <option value="Account 1">
         <option value="Account 2">
         <option value="Account 3">
+        <option value="account-id">
```
Comment (cxfloeder) [inline src/main/webapp/readfile.html:30]: What happens if a user clicks on this option?

File: src/main/webapp/appscript.js [hunk 9]
```diff
@@ -13,27 +13,73 @@
...
 /**
-  * Fetches vendor IDs from VendorServlet and adds them to the
+  * Fetch vendor IDs from VendorServlet and add them to the
   * Customer ID dropdown.
   */
 async function populateCustomerList() {
+  // TODO: Implement functionality for hashmap parsing instead of simple array.
   const response = await fetch('/VendorServlet');
 
+  // VendorServlet returns a hashmap/JSON dictionary.
```
Comment (ianjjohnson) [inline src/main/webapp/appscript.js:23]: rm this comment.

File: src/main/webapp/appscript.js [hunk 10]
```diff
@@ -13,27 +13,73 @@
...
   // Adds the options to the page, allowing them to be viewed. 
   document.getElementById('customer-ids').innerHTML = html;
 }
 
-// TODO: Fetch method for putting the specified config to the page.
+/**
+ * Fetch a json configuration as a string and formats it to be shown on
+ * the readfile page.
+ */
+async function addToPage() {
```
Comment (ianjjohnson) [inline src/main/webapp/appscript.js:45]: Consider renaming this to make it more clear what is being added to the page.

File: src/main/webapp/appscript.js [hunk 11]
```diff
@@ -13,27 +13,73 @@
...
+async function populateEditForm() {
+
+}
 // TODO: Method for enums: Check which of the tags has data under it {isLoggedIn:{loginURL},isLoggedOut:{logooutURL}}}
+async function checkLoginStatus() {
+  
+}
+
+// TODO: Delete config function. /BillingConfig --> doDelete
+async function delteConfiguration() {
```
Comment (ianjjohnson) [inline src/main/webapp/appscript.js:67]: spelling: missing an 'e' in delete

File: src/main/webapp/appscript.js [hunk 12]
```diff
@@ -13,27 +13,73 @@
...
+
+}
+
+// TODO: Create config and redirect/populate edit form. 
+async function createConfiguration() {
+
+}
 
+// function doGetRequest(var endpointUrl, var params) {
```
Comment (ianjjohnson) [inline src/main/webapp/appscript.js:76]: rm all of this commented-out code

File: src/main/webapp/appscript.js [hunk 13]
```diff
@@ -13,27 +13,73 @@
...
+  // Fetch the json configuration and format it to print on the page. 
+  fetch('/BillingConfig')
+    .then(response => response.json())
+    .then(data => document.getElementById('json-text').innerText = 
+    JSON.stringify(data, undefined, 4));
+    
+}
+
 // TODO: Method to populate edit form.
+async function populateEditForm() {
```
Comment (ianjjohnson) [inline src/main/webapp/appscript.js:58]: Does it make sense to split off the javascript for the other pages into other JS files?

i.e. read_file.js, edit_file.js, etc...

File: src/main/webapp/readfile.html [hunk 14]
```diff
@@ -6,7 +6,7 @@
     <link rel="stylesheet" href="readfile.css">
     <script src="appscript.js"></script>
   </head>
-  <body onload="javascript:populateCustomerList()">
```
Comment (ianjjohnson) [inline src/main/webapp/readfile.html:9]: Why remove this? Don't we still need to populate the customer list on load?

File: src/main/webapp/readfile.html (hunk 8 above)
Comment (frostyclawz) [reply src/main/webapp/readfile.html:30]: No support for searching for specific accounts yet.

File: src/main/webapp/readfile.html (hunk 14 above)
Comment (frostyclawz) [reply src/main/webapp/readfile.html:9]: This function doesn't work with the current build of the back-end.

File: .gitignore [hunk 15]
```diff
@@ -26,5 +26,6 @@ target/
 
 # hide client information
 src/main/resources/
+src/main/secrets/
```
Comment (cadecrandall) [inline .gitignore:30]: I was planning on putting this into a separate PR tomorrow. Up to you to keep it

File: src/main/java/servlets/BillingConfig.java [hunk 16]
```diff
@@ -33,7 +33,8 @@
 
 @WebServlet("/BillingConfig")
 public class BillingConfig extends HttpServlet {
-  private static final String CONTENT_TYPE_TEXT_HTML = "text/html;";
+  // May need to be "application/json;
+  private static final String CONTENT_TYPE_APPLICATION_JSON = "application/json;";
```
Comment (cadecrandall) [inline src/main/java/servlets/BillingConfig.java:37]: gj catching this!

File: src/main/java/servlets/OAuthCallbackServlet.java [hunk 17]
```diff
@@ -38,7 +38,7 @@
 @WebServlet("/api/oauth/callback/sheets")
 public class OAuthCallbackServlet extends HttpServlet {
   private static final Logger LOGGER = Logger.getLogger(LoginServlet.class.getName());
-  private final String SECRET_FILEPATH = "../../src/main/resources/secret.txt";
+  private final String SECRET_FILEPATH = "../../src/main/secrets/secret.txt";
```
Comment (cadecrandall) [inline src/main/java/servlets/OAuthCallbackServlet.java:41]: same here for the change in structure. I feel like this should go in a separate PR, but not confident

File: src/main/webapp/readfile.html (hunk 8 above)
Comment (cadecrandall) [reply src/main/webapp/readfile.html:30]: #42

File: src/main/webapp/login.css [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+h1 {
+  text-decoration: underline;
+  width: 500px;
+}
+
+h2 {
+  text-decoration: underline;
+  width: 500px;
```
Comment (cadecrandall) [inline src/main/webapp/login.css:13]: Can you put CSS into src/webapp/assets and scripts into src/webapp/scripts or remove those directories?

File: src/main/java/servlets/OAuthCallbackServlet.java (hunk 17 above)
Comment (frostyclawz) [reply src/main/java/servlets/OAuthCallbackServlet.java:41]: It could. It was more or less a slight change in file path

File: src/main/webapp/appscript.js (hunk 13 above)
Comment (ianjjohnson) [reply src/main/webapp/appscript.js:56]: Still curious about this piece -- should these be in separate files? It may make life easier, especially with Charlie working in parallel on some of the front end stuff

File: src/main/java/servlets/BillingConfig.java [hunk 19]
```diff
@@ -33,7 +33,8 @@
 
 @WebServlet("/BillingConfig")
 public class BillingConfig extends HttpServlet {
-  private static final String CONTENT_TYPE_TEXT_HTML = "text/html;";
+  // May need to be "application/json;
```
Comment (jakecarlson1) [inline src/main/java/servlets/BillingConfig.java:36]: Nit: remove this comment as we are now making the content type json.

File: src/main/webapp/appscript.js (hunk 13 above)
Comment (jakecarlson1) [reply src/main/webapp/appscript.js:56]: Please consider this and reply with your path forward.

File: src/main/webapp/login.css (hunk 18 above)
Comment (jakecarlson1) [reply src/main/webapp/login.css:13]: +1, put CSS into src/webapp/assets and scripts into src/webapp/scripts or remove those directories

File: src/main/webapp/appscript.js (hunk 13 above)
Comment (frostyclawz) [reply src/main/webapp/appscript.js]: Yeah, I'll make separate files for these

Comment (cadecrandall) [reply src/main/webapp/appscript.js]: opened an issue for this #52
